src/ucs/sys/sock.c

```diff
@@ -182,17 +183,23 @@ unsigned ucs_netif_bond_ad_num_ports(const char *bond_name)
     if ((status != UCS_OK) || (ad_num_ports <= 0) ||
         (ad_num_ports > UINT_MAX)) {
         ucs_snprintf_safe(bond_path, PATH_MAX, "%s/%s", UCS_NETIF_CLASS_DIR, bond_name);
-        status = ucs_sys_get_subdir(bond_path, UCS_NETIF_LOWER_BOND_PREFIX,
-                                    sizeof(UCS_NETIF_LOWER_BOND_PREFIX) - 1,
-                                    lower_bond_name, PATH_MAX);
+        status = ucs_sys_get_subdir_with_type(bond_path, DT_LNK, UCS_NETIF_LOWER_BOND_SPECIFIC_PREFIX,
+                                              sizeof(UCS_NETIF_LOWER_BOND_SPECIFIC_PREFIX) - 1,
+                                              lower_bond_name, PATH_MAX);
+        if (status != UCS_OK) {
+            status = ucs_sys_get_subdir_with_type(bond_path, DT_LNK, UCS_NETIF_LOWER_BOND_COMMON_PREFIX,
+                                                  sizeof(UCS_NETIF_LOWER_BOND_COMMON_PREFIX) - 1,
+                                                  lower_bond_name, PATH_MAX);
+        }
+
```

Comment:
@shasson5 Thanks for your comments. I followed your advice to change the code and then did some tests. But looks `ucs_topo_iface_get_sysfs_path` will return NULL and show` /sys/class/net/bond0.192/lower_bond0: sysfs path undetected` though the lower path is resolved in my environment. Need I re-define a new function based ucs_topo_iface_get_sysfs_path but remove the
```
    sysfs_path = ucs_topo_resolve_sysfs_path(ucs_string_buffer_cstr(&dev_path),
                                             path_buffer);
```
It's because mf fix is only to find `/sys/class/net/bond0.192/lower_bond0/bonding/ad_num_ports`

Thanks.